ScreenControl_Unity/Assets/ScreenControl/Service/Scripts/ServiceConnection/ClientConnectionManager.cs

```diff
@@ -38,27 +40,38 @@ private void Awake()
 
         private void OnHandFound()
         {
+            HandPresenceEvent handFoundEvent = new HandPresenceEvent(HandPresenceState.HAND_FOUND);
+
             foreach (ClientConnection _connection in activeConnections)
             {
                 if (_connection.ConnectionState == WebSocketState.Open)
                 {
-                    HandPresenceEvent handFoundEvent = new HandPresenceEvent(HandPresenceState.HAND_FOUND);
-
                     _connection.SendHandPresenceEvent(handFoundEvent);
                 }
             }
+
+            // Cache handPresenceEvent when no clients are connected
+            if (activeConnections.Count == 0)
+            {
+                missedHandPresenceEvent = handFoundEvent;
+            }
         }
         private void OnHandsLost()
         {
+            HandPresenceEvent handsLostEvent = new HandPresenceEvent(HandPresenceState.HANDS_LOST);
+
             foreach (ClientConnection _connection in activeConnections)
             {
                 if (_connection.ConnectionState == WebSocketState.Open)
                 {
-                    HandPresenceEvent handsLostEvent = new HandPresenceEvent(HandPresenceState.HANDS_LOST);
-
                     _connection.SendHandPresenceEvent(handsLostEvent);
                 }
             }
+
+            // Cache handPresenceEvent when no clients are connected
+            if (activeConnections.Count == 0) {
+                missedHandPresenceEvent = handsLostEvent;
+            }
```

Comment:
Should we not always cache the last event sent?

That way if a 2nd client joins, they will know the current state?

Reply:
Mmm that's a good point, though again that's something that starts to support multiple concurrent Clients, which is a use case we've been avoiding explicitly supporting until it can be considered on the whole. 🤔 I think we should leave it as-is for now, but we might want to put together a Confluence page of notes on this topic so that if that work does ever get agreed to that we can remember all these tidbits